.github/workflows/default.yml

```
@@ -1,6 +1,11 @@
 ---
 name: Default Pipeline
 
+permissions:
+  contents: write
+  actions: write
+  pull-requests: write
```

```suggestion
  contents: read
  actions: read
  pull-requests: read
```

Shouldn't we set the least privilege in the default workflow and change them if necessary in the separate jobs!?